src/Optimization/model_internal.jl

```diff
@@ -0,0 +1,94 @@
+mutable struct ModelInternal{T <: AbstractOptimizationContainer}
```

Comment:
Is there a better name for this struct? It kinda seems like a place to tuck away things that didn’t fit elsewhere. `status`, `executions`, and `execution_count` could go into `ModelStatus`. `output_dir`, `recorders`, `console_level`, and `file_level` are user input values. `time_series_cache`…don’t know. `store_params` can go into the store?

Do you agree that some of these feel out of place? Can we reorganize, even in a later PR?

Reply:
If you feel like proposing a re-organization of this I am all for it. At the moment it is the closes thing we have to something that will be re-used in other packages.